File: src/main/java/es/uji/ei1039/agenda/controller/ContactOverview.java
package es.uji.ei1039.agenda.controller;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.Button;

import java.net.URL;
import java.util.ResourceBundle;

public final class ContactOverview {

    @FXML // ResourceBundle that was given to the FXMLLoader
    private ResourceBundle resources;

    @FXML // URL location of the FXML file that was given to the FXMLLoader
    private URL location;

    @FXML // fx:id="newContact"
    private Button newContact; // Value injected by FXMLLoader

    @FXML // fx:id="editContact"
    private Button editContact; // Value injected by FXMLLoader

    @FXML // fx:id="deleteContact"
    private Button deleteContact; // Value injected by FXMLLoader

    @FXML // This method is called by the FXMLLoader when initialization is complete
    void initialize() {
        assert newContact != null : "fx:id=\"newContact\" was not injected: check your FXML file 'ContactOverview.fxml'.";
        assert editContact != null : "fx:id=\"editContact\" was not injected: check your FXML file 'ContactOverview.fxml'.";
        assert deleteContact != null : "fx:id=\"deleteContact\" was not injected: check your FXML file 'ContactOverview.fxml'.";
    }

    @FXML
    void handleNewContact(final ActionEvent event) {
        // TODO
    }

    @FXML
    void handleEditContact(final ActionEvent event) {
        // TODO
    }

    @FXML
    void handleDeleteContact(final ActionEvent event) {
        // TODO
    }
}

File: src/main/kotlin/es/uji/ei1039/agenda/controller/ContactOverview.kt
package es.uji.ei1039.agenda.controller

import es.uji.ei1039.agenda.App
import es.uji.ei1039.agenda.model.Contact
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.scene.control.TableView
import java.net.URL
import java.util.*

class ContactOverview {

    @FXML
    private lateinit var contactsTable: TableView<Contact>

    @FXML
    private lateinit var resources: ResourceBundle

    @FXML
    private lateinit var location: URL

    @FXML
    private lateinit var newContact: Button

    @FXML
    private lateinit var editContact: Button

    @FXML
    private lateinit var deleteContact: Button

    private lateinit var mainApp: App

    @FXML
    private fun initialize() {
        // TODO
    }

    @FXML
    private fun handleNewContact(event: ActionEvent) {
        val tempContact = Contact()
        val onClicked = mainApp.showContactEditDialog(tempContact)
        if (onClicked) {
            //mainApp.getContactData().add(tempContact)
        }
    }

    @FXML
    private fun handleEditContact(event: ActionEvent) {
        // TODO
    }

    @FXML
    private fun handleDeleteContact(event: ActionEvent) {
        // TODO
    }

    fun setMainApp(mainApp : App){
        this.mainApp = mainApp
    }
}
